### DISCUSSION TITLE
Support for Implicit Imports

### DISCUSSION THREAD

**Discussion by we-wake-act21** [eclipse-langium/langium#1341] [Q&A] [answered]
I am working on implementing a language similar to Java, where I aim to offer implicit imports based on the type of grammar for `language1.langium` and `language2.langium`. Specifically, I intend to import all objects from a folder named `Entity`. Additionally, I want to implicitly import all objects of type class at all hierarchical levels within the folder ( `Entity` ) and its sub-folders.

I would appreciate guidance on the best approach to implement this functionality.

**Comment by msujew** [comment, marked as answer]
Hey @we-wake-act21,

when building a language with imports, the way to go is to create a dedicated global scope, see our [own grammar language as an example](https://github.com/eclipse-langium/langium/blob/728b57982b782b31b88ea97aa2a1d3cfe57208b0/packages/langium/src/grammar/references/grammar-scope.ts#L62-L72). Once you have explicit imports available, implicit imports are just an extension of this mechanism. In addition to the explicit imports, you would find which documents in your workspace/stdlib are implicitly imported, and add the exported elements of these documents to the current global scope.
